dask/tests/test_distributed.py

```diff
@@ -659,3 +659,27 @@ async def test_pack_MaterializedLayer_handles_futures_in_graph_properly(c, s, a,
     packed = hlg.__dask_distributed_pack__(c, ["z"], {})
     unpacked = HighLevelGraph.__dask_distributed_unpack__(packed)
     assert unpacked["deps"] == {"x": {fut.key}, "y": {fut.key}, "z": {"y"}}
+
+
+@gen_cluster(client=True)
+async def test_to_sql_engine_kwargs(c, s, a, b):
+    # https://github.com/dask/dask/issues/8738
+    pd = pytest.importorskip("pandas")
+    dd = pytest.importorskip("dask.dataframe")
+    sql = pytest.importorskip("dask.dataframe.io.sql")
+
+    df = pd.DataFrame({"a": range(10), "b": range(10)})
+    ddf = dd.from_pandas(df, npartitions=1)
+    with tmpfile() as f:
+        uri = f"sqlite:///{f}"
+        result = ddf.to_sql(
+            "test", uri, index=True, engine_kwargs={"echo": False}, compute=False
```

Comment:
I see `index=True` here but later in the `assert_eq` we're setting `check_index=False`. I'm wondering if this is intentional and, if so, do we need to set `index=True` here? Pardon my ignorance, I'm not super familiar with `to_sql` / `read_sql_table`

Reply:
Also, is `engine_kwargs` needed?

Reply:
> I see `index=True` here but later in the `assert_eq` we're setting `check_index=False`. I'm wondering if this is intentional and, if so, do we need to set `index=True` here? Pardon my ignorance, I'm not super familiar with `to_sql` / `read_sql_table`

Yeah I'm not very familiar either. I basically just hammered on it until I got something that passed. The only reason I ended up needing `check_index=False` was because the name of the index was different (one was "index", one was None)

> Also, is `engine_kwargs` needed?

Passing engine_kwargs was the motivation for the original work, so I left them in this test to kind of indicate that they work with distributed. That being said I am not asserting anything about the engine_kwargs passing through successfully.